### PR TITLE
Bah versioned sco

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description

As a developer, I need to version the SCO table that is used to display SCO information in the Comparison Tool so that duplicate data is not displayed when new files are uploaded.

database migration done in a separate PR https://github.com/department-of-veterans-affairs/gibct-data-service/pull/454

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19679

front end changes https://github.com/department-of-veterans-affairs/vets-website/pull/10867

## Testing done
1. Run GIDS locally
2. Upload SCO file Multiple times
3. Check SCO table, there should only be one record of an SCO file. 
4. Check Versioned SCO table all previous generated records should be there. 
5. Publish 
6. Check Versioned SCO Archive table and all previous records should be added to this table, SCO table will have current record and Versioned SCO table will have 1 previous record. 

## Acceptance criteria
- [x] When a new preview version is generated, the data is copied from "school_certifying_officials" into a versioned table.
- [x] When a preview version is published, all versions prior to the version being published are archived.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by devin-mccurdy — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

**Comment by jholton** [comment]
The `VersionedSchoolCertifyingOfficial` data is not being used anywhere.
Do you want to use the versioned data in the GIDS responses to the front end (like the way `InstitutionProgram` data is handled)?

File: app/models/versioned_school_certifying_official.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+class VersionedSchoolCertifyingOfficial < ApplicationRecord
```
[the remark this answers is not in the document]
Comment (ksantiagomorgan6) [reply app/models/versioned_school_certifying_official.rb]: The institutionProgram contains data from the edu_programs and programs tables, so it having a unique name makes sense (it’s not just a versioned copy of the programs table, which is what the name VersionedProgram would imply) Going forward all versioned tables will have that naming convention. What do you think?

[the remark this answers is not in the document]
Comment (jholton) [reply app/models/versioned_school_certifying_official.rb]: Ah, I missed that.  Good point.

File: db/migrate/20191002105112_create_versioned_school_certifying_officials.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class CreateVersionedSchoolCertifyingOfficials < ActiveRecord::Migration
```
Comment (jholton) [inline db/migrate/20191002105112_create_versioned_school_certifying_officials.rb]: Please extract the database changes to their own PR.
We'll get get the migrations deployed and executed first.
Then we can deploy the rest of the code in this PR since it depends on those database changes having already occurred.

File: app/models/institution_builder.rb [hunk 3]
```diff
@@ -558,4 +559,33 @@ def self.build_institution_programs(version_number)
     sql = InstitutionProgram.send(:sanitize_sql, [str, version_number])
     InstitutionProgram.connection.execute(sql)
   end
+
+  def self.build_versioned_school_certifying_official(version_number)
+    str = <<-SQL
+    INSERT INTO versioned_school_certifying_officials(facility_code,
```
Comment (jholton) [inline app/models/institution_builder.rb]: Can we clean up the indentation/line breaks of this SQL?
See `build_institution_programs` for a good example.

**Review by delli-gatti-michael — APPROVED** [review]

File: app/models/institution_builder.rb [hunk 4]
```diff
@@ -558,4 +559,34 @@ def self.build_institution_programs(version_number)
     sql = InstitutionProgram.send(:sanitize_sql, [str, version_number])
     InstitutionProgram.connection.execute(sql)
   end
+
+  def self.build_versioned_school_certifying_official(version_number)
+    str = <<-SQL
+      INSERT INTO versioned_school_certifying_officials(facility_code,
```
Comment (jholton) [inline app/models/institution_builder.rb]: Looking better.  Can you move `facility_code` to the line below?

File: app/serializers/institution_profile_serializer.rb [hunk 5]
```diff
@@ -92,7 +92,7 @@ class InstitutionProfileSerializer < ActiveModel::Serializer
   attribute :stem_indicator
   attribute :facility_map
   attribute :programs
-  attribute :school_certifying_officials
+  attribute :versioned_school_certifying_officials
```
Comment (jholton) [inline app/serializers/institution_profile_serializer.rb]: This will change your response json.  Is the frontend expecting `versioned_school_certifying_officials` or `school_certifying_officials`?

Comment (ksantiagomorgan6) [reply app/serializers/institution_profile_serializer.rb]: Yes, I updated the front end to handle that change.

File: app/models/institution_builder.rb (hunk 4 above)
Comment (ksantiagomorgan6) [reply app/models/institution_builder.rb]: Missed that one, I went ahead and updated it.

**Review by jholton — APPROVED** [review]
